middleware/torrentapproval/torrentapproval_test.go

```diff
@@ -0,0 +1,68 @@
+package torrentapproval
+
+import (
+	"context"
+	"encoding/hex"
+	"testing"
+
+	"github.com/chihaya/chihaya/bittorrent"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHandleAnnounce(t *testing.T) {
```

Comment:
Nice!
Now we're going to turn this into a table-driven test. That means we'll define a table of test cases and what should happen in each case.
Something like this:

```go
var cases = []struct{
    cfg Config
    ih string
    approved bool
}{
    // Infohash is whitelisted
    {
        cfg: Config {
            Whitelist: []string{"3532cf2d327fad8448c075b4cb42c8136964a435"},
        },
        ih: "3532cf2d327fad8448c075b4cb42c8136964a435",
        approved: true,
    },//or write a shortened version:
    // Infohash is not whitelisted
    { Config{Whitelist: []string{"3532cf2d327fad8448c075b4cb42c8136964a435"}}, "4532cf2d327fad8448c075b4cb42c8136964a435", false},
    { ....}
}
```

And then you can simplify your test by iterating over the cases. It'd be super nice if you used sub-tests with `t.Run(...)`. You can find an example of what all this looks like in [frontend/udp/connection_id_test.go](https://github.com/chihaya/chihaya/blob/master/frontend/udp/connection_id_test.go).